packages/components/organisms/f-takeawaypay-activation/src/components/TakeawaypayActivation.vue

```diff
@@ -55,12 +68,107 @@ export default {
         homeUrl: {
             type: String,
             required: true
+        },
+        authToken: {
+            type: String,
+            default: ''
         }
     },
     data () {
         return {
-            tenantConfigs
+            tenantConfigs,
+            consumerId: '',
+            consumerEmail: '',
+            activationState: ACTIVATION_STATE_NONE
         };
+    },
+
+    computed: {
+        activationStateComponent () {
```

Comment:
Another option could also be something like
```
activationStateComponent () {
    const store = {
        ACTIVATION_STATE_AVAILABLE_LOGGED_IN: 'activation-logged-in',
        ACTIVATION_STATE_AVAILABLE_NOT_LOGGED_IN: 'activation-not-logged-in',
        ACTIVATION_STATE_SUCCEEDED: 'activation-successful',
    }
    const name = Object.keys(store).find(key => key === this.activationState) ? this.activationState : 'activation-failed'
    const props = Object.assign({},
        (this.activationState === ACTIVATION_STATE_AVAILABLE_LOGGED_IN
            || this.activationState === ACTIVATION_STATE_AVAILABLE_NOT_LOGGED_IN
        ) ? {
            loginUrl: this.loginUrl,
            registrationUrl: this.registrationUrl,
        } : {},
        this.activationState  === 'activation-failed' ? {
            redirectUrl: this.redirectUrl
        } : {},
        this.activationState === ACTIVATION_STATE_AVAILABLE_LOGGED_IN ? {
            activateUrl: this.activateUrl,
            authToken: this.authToken,
            consumerId: this.consumerId,
            consumerEmail: this.consumerEmail
        } : {},
        this.activationState === ACTIVATION_STATE_SUCCEEDED ? {
            homeUrl: this.homeUrl
        } : {}
    )
    return {
        name,
        props: props
    };
},
``` 
to replace all the separate computed properties

Reply:
Based on comment left above you how ever may not want to do this if your opinion is its more clear separately

Reply:
I would prefer existing version as more readable, but it is a matter of taste/habit

Reply:
@matthewhardern yeah, thanks for the idea/example. will probably keep it for now